### DISCUSSION TITLE
complete example full configuration file for the rustdesk server

### DISCUSSION THREAD

**Discussion by OmlineEditor** [rustdesk/rustdesk#8624] [Q&A]
Please give a complete example of all possible variables for the rustdesk server configuration file, the FAQ says that this is just a variable = value, but I do not understand what variables and their possible values are. where can I get the full configuration file for the rustdesk server?

**Comment by DevJupyHUB** [comment]
At your disposal....:

https://github.com/rustdesk/rustdesk-server?tab=readme-ov-file#env-variables
https://github.com/rustdesk/rustdesk-server/discussions/371

**Reply by OmlineEditor** [reply]
> At your disposal....:
> 
> https://github.com/rustdesk/rustdesk-server?tab=readme-ov-file#env-variables [rustdesk/rustdesk-server#371](https://github.com/rustdesk/rustdesk-server/discussions/371)

please explain three parameters:

ALWAYS_USE_RELAY | hbbs | if set to "Y" disallows direct peer connection

RELAY | hbbs | IP address/DNS name of the machines running hbbr (separated by comma)

PORT | hbbs/hbbr | listening port (21116 for hbbs - 21117 for hbbr)


should I put it in quotes or not?
`ALWAYS_USE_RELAY ="Y"`
or
`ALWAYS_USE_RELAY =Y`



how can I specify two values for different ports for one parameter in one configuration file?
`PORT=21116,21117`
how will the program understand what to use and for what?

repeater how can I specify only the IP or can I also specify the port?
`RELAY=192.168.1.1,192.168.2.2`
or
`RELAY=192.168.1.1:11111,192.168.2.2:22222`

**Reply by hugoddjing** [reply]
Just use "PORT=12345", other ports will be generate automatically.

**Reply by igitlin** [reply]
i try ser env ALWAYS_USE_RELAY="Y" and ALWAYS_USE_RELAY=Y
but in log [2025-03-23 00:25:28.491494 +03:00] INFO [src\rendezvous_server.rs:153] ALWAYS_USE_RELAY=N
also i try other key 
LIMIT_SPEED
SINGLE_BANDWIDTH
but it not works
